### PR TITLE
feat: expose the option `checks` to `inputOptions`

### PR DESCRIPTION
**Pull request by shulaoda**
### Description

https://github.com/rolldown/rolldown/blob/225f6a41679fff165eff297ba21578a72b4d88eb/packages/rolldown/tests/fixtures/onwarn/_config.ts#L8

### CODE REVIEW

File: packages/rolldown/src/options/input-options.ts [hunk 1]
```diff
@@ -137,6 +141,7 @@ export interface InputOptions {
   watch?: WatchOptions | false
   dropLabels?: string[]
   keepNames?: boolean
+  checks?: ChecksOptions
```
Comment (hyf0) [inline packages/rolldown/src/options/input-options.ts:148]: Should also add it  in schema.

File: packages/rolldown/src/options/input-options.ts [hunk 2]
```diff
@@ -56,6 +56,10 @@ export interface WatchOptions {
   chokidar?: any
 }
 
+export interface ChecksOptions {
+  circularDependency?: boolean
```
Comment (hyf0) [inline packages/rolldown/src/options/input-options.ts:64]: Add jsdoc

```suggestion
  Wether to emit warnings when detecting circular dependencies.
  @default false
  circularDependency?: boolean
```

File: packages/rolldown/src/options/input-options.ts (hunk 1 above)
Comment (shulaoda) [reply packages/rolldown/src/options/input-options.ts:148]: Related to #2193 
It seems to have already been updated.

File: packages/rolldown/src/binding.d.ts [hunk 3]
```diff
@@ -606,6 +628,25 @@ export interface IsolatedDeclarationsOptions {
   sourcemap?: boolean
 }
 
+export interface IsolatedDeclarationsOptions {
```
Comment (shulaoda) [inline packages/rolldown/src/binding.d.ts:631]: Related to #3058 

I tried what you did on windows, but it didn't work. @hi-ogawa cc

**Review by hyf0 — APPROVED** [review]
Good job.

**Pull request merged**